### PR TITLE
update label for every node in ast

### PR DESCRIPTION
**Pull request by AviAvni**
fixes https://github.com/RedisGraph/RedisGraph/issues/1999

### CODE REVIEW

**Review by swilly22 — APPROVED** [review]

File: tests/flow/test_multi_label.py [hunk 1]
```diff
@@ -208,3 +208,18 @@ def test08_multi_label_variable_length_traversal(self):
         query_result = graph.query(query)
         self.env.assertEquals(query_result.result_set, expected_result)
 
+    def test09_test_query_graph_populate_nodes_labels(self):
+        query = """CREATE (a:L1 {v:0})-[:R1]->()"""
```
Comment (swilly22) [inline tests/flow/test_multi_label.py:212]: Is this create query really necessary ?

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: tests/flow/test_multi_label.py (hunk 1 above)
Comment (AviAvni) [reply tests/flow/test_multi_label.py:212]: yes so we have one node and we want to replace the Label
also it checks that if we found part of the pattern it still merge

File: tests/flow/test_multi_label.py [hunk 2]
```diff
@@ -208,3 +208,18 @@ def test08_multi_label_variable_length_traversal(self):
...
+        self.env.assertEquals(query_result.nodes_created, 2)
+        self.env.assertEquals(query_result.relationships_created, 1)
+
+        # node 'a' is mentioned twice in the following pattern
+        # each time with a different label, when extracting a sub query-graph
+        # we need to make sure all labels mentioned in the extracted pattern
+        # are extracted.
+        query = """MERGE ()-[:R2]->(a:L1)-[:R1]->(a:L2) RETURN *"""
+        query_result = graph.query(query)
+        self.env.assertEquals(query_result.nodes_created, 2)
```
Comment (swilly22) [inline tests/flow/test_multi_label.py:223]: What about `label_created` ?

File: tests/flow/test_multi_label.py (hunk 1 above)
Comment (swilly22) [inline tests/flow/test_multi_label.py:212]: add a comment explaining why we need this query

**Review by swilly22 — CHANGES_REQUESTED** [review]
once both comments address feel free to merge.

**Pull request merged**